### DISCUSSION TITLE
Automatic update

### DISCUSSION THREAD

**Discussion by JordanMess** [YOURLS/YOURLS#3617] [General]
Hello, I always fighting for fresh and actuality.

```
# Edit this file to introduce tasks to be run by cron.
#
# Each task to run has to be defined through a single line
# indicating with different fields when the task will be run
# and what command to run for the task
#
# To define the time you can provide concrete values for
# minute (m), hour (h), day of month (dom), month (mon),
# and day of week (dow) or use '*' in these fields (for 'any').
#
# Notice that tasks will be started based on the cron's system
# daemon's notion of time and timezones.
#
# Output of the crontab jobs (including errors) is sent through
# email to the user the crontab file belongs to (unless redirected).
#
# For example, you can run a backup of all your user accounts
# at 5 a.m every week with:
# 0 5 * * 1 tar -zcf /var/backups/home.tgz /home/
#
# For more information see the manual pages of crontab(5) and cron(8)
#
# m h  dom mon dow   command

0 0 * * * git -C /var/www/html pull && chown -R www-data:www-data /var/www/html && chmod -R 775 /var/www/html
```

Is this is will be clever? Any propositions and ideas please if I would like always fresh version of Yours. Of course I worry if it can fail in not convenient time.

**Comment by dgw** [comment]
I wouldn't recommend blindly pulling new code via cron.

For one, new versions may require running an upgrade script to modify the database schema etc.

For two, the main branch is not guaranteed to be stable and it could just happen to be in a broken state when your automatic pull runs.

**Comment by JordanMess** [comment]
Sounds like idea to implement automatic upgrade of DB if this is required 👍

**Reply by dgw** [reply]
No, because it could run into some error that requires manual intervention, _and_ the user is supposed to make sure they have a backup before running an upgrade in case the upgrade must be rolled back.

**Comment by JordanMess** [comment]
Okay, then question:
Can I manually time to time update my Yourls and be shure that's I am on correct way?

For example I just connect by SSH and do:
```
git -C /var/www/html pull && chown -R www-data:www-data /var/www/html && chmod -R 775 /var/www/html
```
Perhaps also this one, but not shure if it needed for the server life time less then 5 years.
`apt-get update && apt-get upgrade -y`

What do you think? Any Ideaeas you are wellcome. Many thanks.

**Reply by dgw** [reply]
Distribution package updates (`apt-get`) are _definitely_ needed for servers less than 5 years old. You'll get _more_ updates during the early support window than later in a distro version's lifecycle.

I wouldn't update by git pull, ever. [Check if there is a newer release](https://github.com/YOURLS/YOURLS/releases) periodically, maybe add [the Atom feed](https://github.com/YOURLS/YOURLS/releases.atom) to your favorite RSS reader.
